session.go

```diff
@@ -509,8 +523,9 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 				return
 
 			default:
-				// TODO: evaluate
 				debug.Log(1, "RX (Session): unexpected frame: %s\n", body)
+				s.doClose(nil)
```

Comment:
The error passed to `doClose()` is the AMQP `*Error` to send to the peer if we're terminating the session due to an error.  I will add a comment explaining this.

Per [section 2.5.5](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#doc-idp243424), once we send the end performative (END_SENT), the session (and therefore any of its links) can receive frames but not send them (disabling sending will happen in a separate PR).  Previously, we used to just exit the mux after sending the close performative.  However, this could introduce the possibility of dropped frames during shutdown.